File: threedollar-boss-domain/src/main/java/com/depromeet/threedollar/foodtruck/domain/domain/account/repository/BossAccountRepositoryCustom.java
package com.depromeet.threedollar.foodtruck.domain.domain.account.repository;

import com.depromeet.threedollar.foodtruck.domain.domain.account.BossAccount;
import com.depromeet.threedollar.foodtruck.domain.domain.account.BossAccountSocialType;

public interface BossAccountRepositoryCustom {

    BossAccount findBossAccountBySocialTypeAndSocialId(String socialId, BossAccountSocialType socialType);

    BossAccount findBossAccountById(Long bossAccountId);

}

File: threedollar-domain/threedollar-domain-mongo/src/main/kotlin/com/depromeet/threedollar/domain/mongo/domain/bossservice/account/repository/BossAccountRepositoryCustom.kt
package com.depromeet.threedollar.domain.mongo.domain.bossservice.account.repository

interface BossAccountRepositoryCustom {

    fun existsBossAccountById(id: String): Boolean

    fun findBossAccountById(id: String): com.depromeet.threedollar.domain.mongo.domain.bossservice.account.BossAccount?

    fun findBossAccountBySocialInfo(socialId: String, socialType: com.depromeet.threedollar.domain.mongo.domain.bossservice.account.BossAccountSocialType): com.depromeet.threedollar.domain.mongo.domain.bossservice.account.BossAccount?

    fun existsBossAccountBySocialInfo(socialId: String, socialType: com.depromeet.threedollar.domain.mongo.domain.bossservice.account.BossAccountSocialType): Boolean

}
